ts/Core/Axis/OrdinalAxis.ts

```diff
@@ -845,6 +852,91 @@ namespace OrdinalAxis {
             return groupPositions;
         };
 
+        /**
+         * Get index of point inside the extended ordinal positions array.
+         *
+         * @private
+         * @function Highcharts.Axis#getIndexOfPoint
+         *
+         * @param {number} val
+         *        The pixel value of a point.
+         *
+         * @param {Array<number>} [ordinalPositions]
+         *        An array of points visible in the current plot area.
+         *
+         * @param {Array<number>} [extendedOrdinalPositions]
+         *        An array of all points available on the axis
+         *        for the given data set.
+         *
+         * @return {number}
+         */
+        axisProto.getIndexOfPoint = function (
+            val: number,
+            ordinalPositions: Array<number>,
+            extendedOrdinalPositions: Array<number>
+        ): number {
+            const axis = this,
+                // Distance in pixels between two points
+                // on the ordinal axis in the current zoom.
+                ordinalPointPixelInterval = axis.len / ordinalPositions.length - 1,
+                shiftIndex = val / ordinalPointPixelInterval;
+
+            // Make sure that the returned index does not exceed array
+            // boundaries. If so, return the first or the last index of array.
+            return Math.min(
+                Math.max(
+                    extendedOrdinalPositions.indexOf(axis.min as any) + shiftIndex,
+                    0
+                ), // first index
+                extendedOrdinalPositions.length - 1 // last index
+            );
+        };
+
+        /**
+         * Get axis position of given index of the extended ordinal positions.
+         * Used only when panning an ordinal axis.
+         *
+         * @private
+         * @function Highcharts.Axis#lin2val
+         *
+         * @param {number} index
+         *        The index value of searched point
+         *
+         * @return {number}
+         */
+        axisProto.index2val = function (index: number): number {
+            let axis = this,
+                ordinal = axis.ordinal,
+                // Context could be changed to extendedOrdinalPositions.
+                ordinalPositions = ordinal.positions;
+
+            // The visible range contains only equally spaced values.
+            if (!ordinalPositions) {
+                return index; // TODO Find better return value if ordinal doesn't exist.
```

Comment:
is it even possible that ordinalPositions is undefined?

Reply:
This method is not public, so the risk that the `ordinalPositions` doesn't exist are small. 
However for now this condition might be pointless because I've changed what is passed to that method. 
And it won't prevent from crashing.
I think we might remove that additional if statement? 
@pawelfus could you please advise?

Part changed in the `Axis.ts`:
``` TS
returnValue = axis.lin2val(axis.logarithmic ? returnValue : val);
```